source/Octopus.Tentacle/Commands/RunAgentCommand.cs

```diff
@@ -74,6 +97,74 @@ public class RunAgentCommand : AbstractStandardCommand
                 // This option is added to show help
             });
         }
+#if !NETFRAMEWORK
+        private IHostBuilder CreateHostBuilder() =>
+            Host.CreateDefaultBuilder()
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseKestrel(options =>
+                    {
+                        // options.ConfigureHttpsDefaults(httpsOptions =>
+                        // {
+                        //     httpsOptions.ClientCertificateMode = ClientCertificateMode.RequireCertificate;
+                        //     httpsOptions.ClientCertificateValidation = (certificate, _, __) =>
+                        //     {
+                        //         using var cert = new X509Certificate2(certificate.Export(X509ContentType.Cert));
+                        //         return configuration.Value.TrustedOctopusServers.Any(serverConfiguration => serverConfiguration.Thumbprint == cert.Thumbprint);
+                        //     };
+                        // });
+                        options.Limits.MaxRequestBodySize = null;
+
+                        options.ListenAnyIP(5001, listenOptions =>
+                        {
+                            listenOptions.Protocols = HttpProtocols.Http1;
+                            // listenOptions.UseHttps(configuration.Value.TentacleCertificate!);
+                        });
+                        options.ListenAnyIP(5002, listenOptions =>
+                        {
+                            listenOptions.Protocols = HttpProtocols.Http2;
+                            // listenOptions.UseHttps(configuration.Value.TentacleCertificate!);
+                        });
+                    });
+                    webBuilder.UseStartup<Startup>();
+                }).UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory(container));
```

Comment:
Had to update Autofac so I could easily get the existing container services into the Kestrel's container to allow for DI in the gRPC service